### PR TITLE
Got rid of old spring-social dependency 

### PR DESCRIPTION
**Pull request by MarekB01**
Related to: https://github.com/BroadleafCommerce/QA/issues/4974

Removed spring-social dependency as it became deprecated. Rewrote BroadleafSocialRegisterController to use OAuth2 implementation.

### CODE REVIEW

**Comment by riteshadhikari17** [comment]
This is the error in jenkins
```
[ERROR]     'dependencies.dependency.version' for org.springframework.security:spring-security-oauth2-client:jar is missing. @ line 101, column 21
```

FYI Module Parent is for modules. Core does not use MP. Define the version here and close the PR for MP.

**Comment by MarekB01** [comment]
Now it can be merged

File: core/broadleaf-profile-web/pom.xml [hunk 1]
```diff
@@ -81,5 +81,11 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-oxm</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-client</artifactId>
+            <version>5.5.3</version>
```
Comment (riteshadhikari17) [inline core/broadleaf-profile-web/pom.xml]: Don't define version here. It should come from root pom.

File: pom.xml [hunk 2]
```diff
@@ -1162,6 +1162,11 @@
                 <artifactId>spring-boot-autoconfigure</artifactId>
                 <version>${spring.boot.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.springframework.security</groupId>
+                <artifactId>spring-security-oauth2-client</artifactId>
+                <version>6.0.2</version>
```
Comment (riteshadhikari17) [inline pom.xml]: This version needs to sync with spring security. We define it as property above so use it.
`${spring.security.version}`. Also shouldn't you remove the spring-social dependency below?

File: core/broadleaf-framework-web/pom.xml [hunk 3]
```diff
@@ -98,5 +98,9 @@
             <groupId>org.easymock</groupId>
             <artifactId>easymockclassextension</artifactId>
         </dependency>
+        <dependency>
```
Comment (oleksiimiroshnyk) [inline core/broadleaf-framework-web/pom.xml:97]: I see you've added a new dependency, but what about deleting old spring social ? Is it still there ?

**Review by oleksiimiroshnyk — CHANGES_REQUESTED** [review]

**Review by riteshadhikari17 — APPROVED** [review]

**Review by oleksiimiroshnyk — APPROVED** [review]

**Pull request merged**